### PR TITLE
Clear database layer of remaining triggers - Closes #1495, #1450

### LINKED ISSUE #1495: Clear database layer of remaining triggers
### Expected behavior
There should be no logic in database. Replace triggers with SQL queries.

- [x] Table `rounds_fees` should be refactored to cointain also rewards and triggers should be replaced with SQL query/queries executed as part of rounds promises chain.
- [x] Review if there are some other triggers.

### Actual behavior
There is table `rounds_fees` that is populated by triggers.

### Steps to reproduce
N/A

### Which version(s) does this affect? (Environment, OS, etc...)
1.0.0

### PR DESCRIPTION
**Pull request by 4miners**
### What was the problem?
Check referenced issues.

### How did I fix it?
- Replace `rounds_fees` table with `rounds_rewards`
- Populate `rounds_rewards` table with blockchain data
- Apply round exceptions to `rounds_rewards`
- Add queries for maintaining `rounds_rewards` to `applyRound` and `backwardLand`
- Fix query `aggregateBlocksReward` to use new table

### How to test it?
Manually, test will come as part of functional system tests for rounds.

### Review checklist

* The PR solves https://github.com/LiskHQ/lisk/issues/1495, https://github.com/LiskHQ/lisk/issues/1450
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: db/repos/rounds.js [hunk 1]
```diff
@@ -230,6 +230,38 @@ class RoundsRepository {
...
+	 * Delete information about entire round rewards from rounds_rewards.
+	 *
+	 * @param {Number} round - Round number
+	 * @return {Promise}
+	 */
+	deleteRoundRewards(round) {
+		return this.db.none(sql.deleteRoundRewards, {
+			round,
+		});
+	}
```
Comment (karmacoma) [inline db/repos/rounds.js]: Assuming unit tests will be written here: https://github.com/LiskHQ/lisk/issues/1484

File: db/sql/migrations/updates/20180205000003_create_rounds_rewards_table.sql [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+DROP TRIGGER  IF EXISTS rounds_fees_insert ON "blocks";
+DROP FUNCTION IF EXISTS round_fees_insert();
+
+-- Create table 'rounds_rewards' for storing rewards
+CREATE TABLE IF NOT EXISTS "rounds_rewards"(
+	"timestamp" INT    NOT NULL,
+	"fees"      BIGINT NOT NULL,
+	"reward"    BIGINT NOT NULL,
+	"round"     INT    NOT NULL,
+	"pk"        BYTEA  NOT NULL
```
Comment (karmacoma) [inline db/sql/migrations/updates/20180205000003_create_rounds_rewards_table.sql]: Can we stick to `"publicKey"` please?

File: db/sql/migrations/updates/20180205000003_create_rounds_rewards_table.sql [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+-- Compute all rewards for previous rounds and insert them to 'rounds_rewards'
+DO $$
+	DECLARE
+		row record;
+	BEGIN
+		RAISE NOTICE 'Calculating rewards for rounds, please wait...';
+		FOR row IN
+			SELECT
+				-- Round number
+				CEIL(height / 101::float)::int AS round
```
Comment (karmacoma) [inline db/sql/migrations/updates/20180205000003_create_rounds_rewards_table.sql]: Lower case function calls please, as per @vitaly-t standard.

File: db/sql/migrations/updates/20180205000003_create_rounds_rewards_table.sql [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+					SELECT
+						b.timestamp, b.height, b."generatorPublicKey" AS pk, b."totalFee" AS fees,
+						b.reward AS reward
+					FROM blocks b
+					WHERE CEIL(b.height / 101::float)::int = row.round AND b.height > 1
+				),
+				-- Calculating total fees of round
+				fees AS (SELECT SUM(fees) AS total, FLOOR(SUM(fees) / 101) AS single FROM round),
+				-- Get last delegate and timestamp of round's last block
+				last AS (SELECT pk, timestamp FROM round ORDER BY height DESC LIMIT 1)
```
Comment (karmacoma) [inline db/sql/migrations/updates/20180205000003_create_rounds_rewards_table.sql]: Again `"publicKey"` over `"pk"`, as more descriptive.

File: db/sql/migrations/updates/20180205000004_apply_round_exception_mainnet.sql [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  DESCRIPTION: Apply reward exceptions for round 27040 (only on mainnet).
+
+  PARAMETERS: None
+*/
+
+DO $$
+	DECLARE
+		is_mainnet INT;
+	BEGIN
+		SELECT COUNT(1) FROM blocks WHERE height = 1 AND ENCODE("generatorPublicKey", 'hex') = 'd121d3abf5425fdc0f161d9ddb32f89b7750b4bdb0bff7d18b191d4b4bafa6d4' INTO is_mainnet;
```
Comment (karmacoma) [inline db/sql/migrations/updates/20180205000004_apply_round_exception_mainnet.sql]: Lower case function calls please, as per @vitaly-t standard.

File: db/sql/migrations/updates/20180205000004_apply_round_exception_mainnet.sql [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+					SELECT
+						b.timestamp, b.height, b."generatorPublicKey" AS pk, b."totalFee" * 2 AS fees,
+						b.reward * 2 AS reward, 10000000 AS fb
+					FROM blocks b
+					WHERE CEIL(b.height / 101::float)::int = 27040
+				),
+				-- Calculating total fees of round, apply exception fees bonus
+				fees AS (SELECT SUM(fees) + fb AS total, FLOOR((SUM(fees) + fb) / 101) AS single FROM round GROUP BY fb),
+				-- Get last delegate and timestamp of round's last block
+				last AS (SELECT pk, timestamp FROM round ORDER BY height DESC LIMIT 1)
```
Comment (karmacoma) [inline db/sql/migrations/updates/20180205000004_apply_round_exception_mainnet.sql]: Again `"publicKey"` over `"pk"`, as more descriptive.

File: logic/round.js [hunk 7]
```diff
@@ -210,6 +212,21 @@ Round.prototype.restoreVotesSnapshot = function() {
...
+ * @return {function} Promise
+ */
+Round.prototype.deleteRoundRewards = function() {
+	this.scope.library.logger.debug(
+		`Deleting rewards for round ${this.scope.round}`
+	);
+	return (this.t || this.scope.library.db).rounds.deleteRoundRewards(
+		this.scope.round
+	);
+};
```
Comment (karmacoma) [inline logic/round.js]: Add unit test coverage for: `Round.prototype.deleteRoundReward`

File: logic/round.js [hunk 8]
```diff
@@ -308,10 +337,35 @@ Round.prototype.applyRound = function() {
...
+				item.round,
+				item.pk
+			)
+		);
+	});
+
+	self.scope.library.logger.trace('Applying round', {
+		queries_count: queries.length,
+		rewards: roundRewards,
+	});
```
Comment (karmacoma) [inline logic/round.js]: Updated unit test coverage for: `Round.prototype.applyRound`

File: test/unit/logic/round.js [hunk 9]
```diff
@@ -590,6 +591,28 @@ describe('rounds', () => {
...
+		});
+
+		it('should return promise', () => {
+			expect(isPromise(res)).to.be.true;
+		});
+
+		it('query should be called with no args', () => {
+			return res.then(res => {
+				expect(res).to.equal('success');
+				expect(stub.calledWith()).to.be.true;
```
Comment (karmacoma) [inline test/unit/logic/round.js]:
```js
expect(stub).to.have.been.calledWith();
```

See: https://github.com/domenic/sinon-chai#assertions

**Comment by 4miners** [comment]
@karmacoma Requested changes done, please review once again.

**Pull request merged**